speakers/src/main/java/in/droidcon/speakers/domain/SpeakerRepositoryImplementation.kt

```diff
@@ -1,49 +1,45 @@
 package `in`.droidcon.speakers.domain
 
-import `in`.droidcon.base.core.getOneSpeaker
-import `in`.droidcon.base.core.getSpeakerList
-import `in`.droidcon.data.speakers.model.SpeakerEntity
-import com.google.firebase.firestore.FirebaseFirestore
+import `in`.droidcon.base.repo.remote.ServiceFactory
+import `in`.droidcon.speakers.BuildConfig
+import `in`.droidcon.speakers.model.SpeakerEntity
+import `in`.droidcon.speakers.remote.SpeakerMapper
+import `in`.droidcon.speakers.remote.SpeakerService
 import io.reactivex.Single
 
 /**
  * @author Adnan A M
  * @since  20/03/19
  */
-class SpeakerRepositoryImplementation: SpeakerRepository {
+class SpeakerRepositoryImplementation : SpeakerRepository {
 
-    private val fireStore: FirebaseFirestore = FirebaseFirestore.getInstance()
+    //TODO: DI with Koin
```

Comment:
Can anyone help me with Koin? I couldn't figure out how to `inject` in non-Android files

Reply:
@adnan-SM @Hariofspades @brijeshmasrani @iamBedant

Reply:
You can implement `KoinComponent` interface and you would be able to inject on any class as long as the app has called startKoin earlier.

Reply:
@RivuChk - For Koin to work with Non Android classes, there is an interface called KoinComponent which you can just use. 

More info here :

https://github.com/InsertKoinIO/koin/issues/139
https://insert-koin.io/docs/2.0/documentation/reference/index.html#_create_a_koin_component

But ya, short answer just add `:KoinComponent` to the class declaration and you can inject !

Reply:
Thanks @adnan-SM and @vivek1794 it worked